_docs/step-development-guideline.md

```diff
@@ -1,5 +1,45 @@
 # Step Development Guideline
 
+## Step dependencies
+
+In the context of Steps, dependency means an executable available in the PATH environment variable, which is executed by the step.
+
+`deps` should not include an executable needed to run the step.
+
+_For example `go` should not be listed as a step dependency in case of a step written in go, but needs to be listed if the step calls the `go` binary as the [Go test step](https://github.com/bitrise-steplib/steps-go-test/blob/master/main.go#L86) does._
+
+A step dependency is installed by the `Bitrise CLI` if it is not available in the PATH environment variable. Unused dependency increases the build time unnecessarily, avoid listing a dependency which is not executed by the step.
+
+Since steps can be performed in any environment where Bitrise CLI can run, list every used dependency, even if you know that they are pre-installed on the Bitrise build virtual machines.
```

Comment:
Side note:
It impiles that this info: https://discuss.bitrise.io/t/golang-as-a-dependency/6989/8 is no longer valid. Moreover, already removed `git` dependencies should be restored. AFAIK this one is affected: https://github.com/bitrise-io/bitrise-steplib/pull/1894

Reply:
Sorry about that we are working on to make up our missing documentation.